src/auth/interface.js

```diff
@@ -0,0 +1,253 @@
+ * `initialize` function.
+ *
+ * @event
+ * @see {@link module:Initialization~initialize}
+ */
+export const AUTHENTICATED_USER_CHANGED = `${AUTHENTICATED_USER_TOPIC}.CHANGED`;
+
+const optionsShape = {
+  config: PropTypes.shape({
+    BASE_URL: PropTypes.string.isRequired,
+    LMS_BASE_URL: PropTypes.string.isRequired,
+    LOGIN_URL: PropTypes.string.isRequired,
+    LOGOUT_URL: PropTypes.string.isRequired,
+    REFRESH_ACCESS_TOKEN_ENDPOINT: PropTypes.string.isRequired,
+    ACCESS_TOKEN_COOKIE_NAME: PropTypes.string.isRequired,
+    CSRF_TOKEN_API_PATH: PropTypes.string.isRequired,
+  }).isRequired,
+  loggingService: PropTypes.shape({
+    logError: PropTypes.func.isRequired,
+    logInfo: PropTypes.func.isRequired,
+  }).isRequired,
+};
+
+const serviceShape = {
+  getAuthenticatedHttpClient: PropTypes.func.isRequired,
+  getHttpClient: PropTypes.func.isRequired,
+  getLoginRedirectUrl: PropTypes.func.isRequired,
+  redirectToLogin: PropTypes.func.isRequired,
+  getLogoutRedirectUrl: PropTypes.func.isRequired,
+  redirectToLogout: PropTypes.func.isRequired,
+  getAuthenticatedUser: PropTypes.func.isRequired,
+  setAuthenticatedUser: PropTypes.func.isRequired,
+  fetchAuthenticatedUser: PropTypes.func.isRequired,
+  ensureAuthenticatedUser: PropTypes.func.isRequired,
+  hydrateAuthenticatedUser: PropTypes.func.isRequired,
+};
+
+let service;
+
+/**
+ *
+ * @param {class} AuthService
+ * @param {*} options
+ * @returns {AuthService}
+ */
+export function configure(AuthService, options) {
+  PropTypes.checkPropTypes(optionsShape, options, 'property', 'Auth');
+  service = new AuthService(options);
+  PropTypes.checkPropTypes(serviceShape, service, 'property', 'AuthService');
+  return service;
+}
+
+/**
+ *
+ *
+ * @returns {AuthService}
+ */
+export function getAuthService() {
+  if (!service) {
+    throw Error('You must first configure the auth service.');
+  }
+
+  return service;
+}
+
+/**
+ *
+ */
+export function resetAuthService() {
+  service = null;
+}
+
+/**
+ *
+ */
+export function getAuthenticatedHttpClient() {
+  return service.getAuthenticatedHttpClient();
+}
+
+/**
+ *
+ */
+export function getHttpClient() {
+  return service.getHttpClient();
+}
+
+/**
+ *
+ */
+export function getLoginRedirectUrl(redirectUrl) {
+  return service.getLoginRedirectUrl(redirectUrl);
+}
+
+/**
+ *
+ */
+export function redirectToLogin(redirectUrl) {
+  return service.redirectToLogin(redirectUrl);
+}
+
+/**
+ *
+ */
+export function getLogoutRedirectUrl(redirectUrl) {
+  return service.getLogoutRedirectUrl(redirectUrl);
+}
+
+/**
+ *
+ */
+export function redirectToLogout(redirectUrl) {
+  return service.redirectToLogout(redirectUrl);
+}
+
+/**
+ *
+ */
+export function getAuthenticatedUser() {
+  return service.getAuthenticatedUser();
+}
+
+/**
+ *
+ */
+export function setAuthenticatedUser(authUser) {
+  service.setAuthenticatedUser(authUser);
+  publish(AUTHENTICATED_USER_CHANGED);
```

Comment:
Note how this `publish` call is part of the interface.  This is a choice we made that the interface can provide extra functionality like this, which allows the implementation to be agnostic and more unaware of larger concerns of the frontend-platform as a whole.